backend/server.py

```
@@ -1,276 +1,511 @@
-        generated_answer = qa_response["answer"]
+    input_text, err = _validate_input_text(input_text)
+    if err:
+        return err
 
-        # Calculate similarity between generated answer and each option
-        options_with_answer = options + [generated_answer]
-        vectorizer = TfidfVectorizer().fit_transform(options_with_answer)
-        vectors = vectorizer.toarray()
-        generated_answer_vector = vectors[-1].reshape(1, -1)
+    input_questions, err_q = _validate_string_list(data.get("input_question"), "input_question")
+    if err_q: return err_q
```

Ruff (0.15.10)

[error] Multiple statements on one line (colon)

(E701)